// github-embed/npm/components/FrameHTML/FrameHTMLView.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FrameHTMLView = function FrameHTMLView(_ref) {
    var shown = _ref.shown,
        loaded = _ref.loaded,
        src = _ref.url,
        _onLoad = _ref.onLoad;
    return _react2.default.createElement('iframe', _extends({}, loaded ? { src: src } : {}, {
        style: {
            display: shown ? '' : 'none'
        },
        onLoad: function onLoad() {
            return _onLoad();
        }
    }));
};

FrameHTMLView.propTypes = {
    shown: _react.PropTypes.bool.isRequired,
    loaded: _react.PropTypes.bool.isRequired,
    url: _react.PropTypes.string.isRequired,
    onLoad: _react.PropTypes.func.isRequired
};

var _default = FrameHTMLView;
exports.default = _default;
;

var _temp = function () {
    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
        return;
    }

    __REACT_HOT_LOADER__.register(FrameHTMLView, 'FrameHTMLView', 'src/components/FrameHTML/FrameHTMLView.js');

    __REACT_HOT_LOADER__.register(_default, 'default', 'src/components/FrameHTML/FrameHTMLView.js');
}();

;
