// github-embed/npm/components/FrameHTML/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _reactRedux = require('react-redux');

var _actions = require('../../actions');

var actions = _interopRequireWildcard(_actions);

var _FrameHTMLView = require('./FrameHTMLView');

var _FrameHTMLView2 = _interopRequireDefault(_FrameHTMLView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function mapState(_ref, _ref2) {
    var frames = _ref.frames;
    var index = _ref2.index;
    var _frames$index = frames[index],
        shown = _frames$index.shown,
        loaded = _frames$index.loaded,
        url = _frames$index.url;


    return {
        shown: shown,
        loaded: loaded,
        url: url
    };
}

function mapDispatch(dispatch) {
    return {
        onLoad: function onLoad() {
            return dispatch(actions.setLoaded(true));
        }
    };
}

var _default = (0, _reactRedux.connect)(mapState, mapDispatch)(_FrameHTMLView2.default);

exports.default = _default;
;

var _temp = function () {
    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
        return;
    }

    __REACT_HOT_LOADER__.register(mapState, 'mapState', 'src/components/FrameHTML/index.js');

    __REACT_HOT_LOADER__.register(mapDispatch, 'mapDispatch', 'src/components/FrameHTML/index.js');

    __REACT_HOT_LOADER__.register(_default, 'default', 'src/components/FrameHTML/index.js');
}();

;
