// github-embed/npm/components/Header/HeaderView.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HeaderView = function HeaderView(_ref) {
    var items = _ref.items,
        githubLink = _ref.githubLink,
        onLabelClick = _ref.onLabelClick;
    return _react2.default.createElement(
        'div',
        { className: 'github-embed-nav' },
        items.map(function (_ref2, index) {
            var shown = _ref2.shown,
                label = _ref2.label;
            return _react2.default.createElement(
                'a',
                {
                    className: (0, _classnames2.default)({
                        'github-embed-nav-link': true,
                        'github-embed-nav-link-shown': shown
                    }),
                    key: index,
                    onClick: function onClick() {
                        return onLabelClick(index);
                    }
                },
                label
            );
        }),
        _react2.default.createElement(
            'a',
            {
                className: 'github-embed-nav-on-github',
                target: '_blank',
                rel: 'noopener noreferrer',
                href: githubLink
            },
            'On Github'
        )
    );
};

HeaderView.propTypes = {
    items: _react.PropTypes.arrayOf(_react.PropTypes.shape({
        label: _react.PropTypes.string.isRequired,
        shown: _react.PropTypes.bool.isRequired
    })),
    onLabelClick: _react.PropTypes.func.isRequired,
    githubLink: _react.PropTypes.string.isRequired
};

var _default = HeaderView;
exports.default = _default;
;

var _temp = function () {
    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
        return;
    }

    __REACT_HOT_LOADER__.register(HeaderView, 'HeaderView', 'src/components/Header/HeaderView.js');

    __REACT_HOT_LOADER__.register(_default, 'default', 'src/components/Header/HeaderView.js');
}();

;
