// github-embed/npm/components/Header/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _reactRedux = require('react-redux');

var _actions = require('../../actions');

var actions = _interopRequireWildcard(_actions);

var _HeaderView = require('./HeaderView');

var _HeaderView2 = _interopRequireDefault(_HeaderView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function mapState(state) {
    var _ref = state.settings || {},
        owner = _ref.owner,
        repo = _ref.repo,
        ref = _ref.ref;

    return {
        items: state.frames.map(function (item) {
            return {
                label: item.label || item.path,
                shown: item.shown
            };
        }),
        githubLink: 'http://github.com/' + owner + '/' + repo + '/tree/' + ref + '/'
    };
}

function mapDispatch(dispatch) {
    return {
        onLabelClick: function onLabelClick(index) {
            return dispatch(actions.activateFrame(index));
        }
    };
}

var _default = (0, _reactRedux.connect)(mapState, mapDispatch)(_HeaderView2.default);

exports.default = _default;
;

var _temp = function () {
    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
        return;
    }

    __REACT_HOT_LOADER__.register(mapState, 'mapState', 'src/components/Header/index.js');

    __REACT_HOT_LOADER__.register(mapDispatch, 'mapDispatch', 'src/components/Header/index.js');

    __REACT_HOT_LOADER__.register(_default, 'default', 'src/components/Header/index.js');
}();

;
