// github-embed/npm/constants/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var SET_LOADED = exports.SET_LOADED = 'SET_LOADED';
var ERROR = exports.ERROR = 'ERROR';
var EMBED = exports.EMBED = 'EMBED';
var SET_CODE_CONTENT = exports.SET_CODE_CONTENT = 'SET_CODE_CONTENT';
var LOAD_FRAME = exports.LOAD_FRAME = 'LOAD_FRAME';
var SHOW_FRAME = exports.SHOW_FRAME = 'SHOW_FRAME';
var SET_SETTINGS = exports.SET_SETTINGS = 'SET_SETTINGS';
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SET_LOADED, 'SET_LOADED', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(ERROR, 'ERROR', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(EMBED, 'EMBED', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(SET_CODE_CONTENT, 'SET_CODE_CONTENT', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(LOAD_FRAME, 'LOAD_FRAME', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(SHOW_FRAME, 'SHOW_FRAME', 'src/constants/index.js');

  __REACT_HOT_LOADER__.register(SET_SETTINGS, 'SET_SETTINGS', 'src/constants/index.js');
}();

;
