// github-embed/npm/reducers/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = application;

var _constants = require('../constants');

var initialState = {
    settingsPathData: null,
    loaded: false,
    error: null,
    lineNumbers: null,
    frames: []
};

function application() {
    var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : initialState;
    var action = arguments[1];

    switch (action.type) {
        case _constants.SET_LOADED:
            {
                var loaded = action.loaded;

                return _extends({}, state, {
                    loaded: loaded
                });
            }
        case _constants.ERROR:
            {
                var error = action.error;

                return _extends({}, state, {
                    error: error
                });
            }
        case _constants.SET_SETTINGS:
            {
                var settings = action.settings;

                return _extends({}, state, {
                    settings: settings
                });
            }
        case _constants.EMBED:
            {
                var _state$settings = state.settings,
                    _state$settings$lineN = _state$settings.lineNumbers,
                    lineNumbers = _state$settings$lineN === undefined ? true : _state$settings$lineN,
                    embed = _state$settings.embed;

                return _extends({}, state, {
                    lineNumbers: lineNumbers,
                    frames: embed.map(function (item) {
                        var newValue = _extends({
                            loaded: false,
                            shown: false
                        }, item);

                        if (item.type !== 'htmlpage') {
                            var splittedPath = item.path.split('.');
                            newValue.type = item.type || splittedPath[splittedPath.length - 1];
                            newValue.code = '';
                        }

                        return newValue;
                    })
                });
            }
        case _constants.LOAD_FRAME:
            {
                return _extends({}, state, {
                    frames: state.frames.map(function (item, index) {
                        return _extends({}, item, {
                            loaded: action.index === index ? true : item.loaded
                        });
                    })
                });
            }
        case _constants.SHOW_FRAME:
            {
                return _extends({}, state, {
                    frames: state.frames.map(function (item, index) {
                        return _extends({}, item, {
                            shown: action.index === index
                        });
                    })
                });
            }
        case _constants.SET_CODE_CONTENT:
            {
                return _extends({}, state, {
                    frames: state.frames.map(function (item, index) {
                        return _extends({}, item, {
                            code: action.index === index ? action.code : item.code
                        });
                    })
                });
            }
        default:
            {
                return state;
            }
    }
}
;

var _temp = function () {
    if (typeof __REACT_HOT_LOADER__ === 'undefined') {
        return;
    }

    __REACT_HOT_LOADER__.register(initialState, 'initialState', 'src/reducers/index.js');

    __REACT_HOT_LOADER__.register(application, 'application', 'src/reducers/index.js');
}();

;
